### PR TITLE
Generate labels, releases, and milestones client

### PR DESCRIPTION
**Pull request by hmharvey**
Part of #1157. Generalizes the `openapi_client_generator` to include the `labels`, `releases`, and `milestones` clients. 

Some [breaking] changes: 
* Labels: `add_label` is now `create_label` and has no default color
* Milestones: removing the `list_milestone` alias, adding the method`milestone_labels` from `labels_for_milestone` in the labels client
* Releases: `release`, `update_release`, `delete_release`, `release_assets` all switch from using a url parameter to `repo` and `release_id`. Likewise, `release_asset`, `update_release_asset`, and `delete_release_asset` all switch from using a url to `repo` and `asset_id`. `release_for_tag` is also now `release_by_tag`

Also retroactively changes some comments to fix yard guidelines.

TODO:
- [x] discuss `label` definitions for issues [they'll move to be under "issues"]
- [x] investigate `upload_asset` [we'll wait until it's officially apart of openAPI]
- [x] clean up testing for assets

### CODE REVIEW

File: lib/octokit/client/deployments.rb [hunk 1]
```diff
@@ -18,10 +18,10 @@ def deployment(repo, deployment_id, options = {})
       # List deployments
       #
       # @param repo [Integer, String, Repository, Hash] A GitHub repository
-      # @param options [String] :sha The SHA recorded at creation time.
-      # @param options [String] :ref The name of the ref. This can be a branch, tag, or SHA.
-      # @param options [String] :task The name of the task for the deployment (e.g., `deploy` or `deploy:migrations`).
-      # @param options [String] :environment The name of the environment that was deployed to (e.g., `staging` or `production`).
+      # @option options [String] :sha The SHA recorded at creation time.
```
Comment (tarebyte) [inline lib/octokit/client/deployments.rb]: 🎉

File: lib/octokit/client/deployments.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ module Deployments
       #
       # @param repo [Integer, String, Repository, Hash] A GitHub repository
       # @param deployment_id [Integer] The ID of the deployment
-      # @return <Sawyer::Resource> A single deployment
+      # @return [Sawyer::Resource] A single deployment
```
Comment (tarebyte) [inline lib/octokit/client/deployments.rb]: 🎉

File: lib/openapi_client_generator.rb [hunk 3]
```diff
@@ -62,7 +62,11 @@ def #{method_name}(#{parameters})
     end
 
     def singular?
-      (definition.path.path.split("/").last.include? "id") || (definition.summary.include? " a ")
+      if definition.responses.first.content.present?
+        definition.responses.first.content["application/json"]["schema"]["type"] != "array"
+      else
+        true
+      end
```
Comment (tarebyte) [inline lib/openapi_client_generator.rb]: WDYT about

```ruby
def singular?
  return true unless definition.responses.first.content.present?
  definition.responses.first.content["application/json"]["schema"]["type"] != "array"
end
```

Comment (hmharvey) [reply lib/openapi_client_generator.rb]: Much better 👍

**Pull request merged**